Adyen/Core/Components/PresentableInitiableComponent.swift

```diff
@@ -0,0 +1,12 @@
+//
+// Copyright (c) 2024 Adyen N.V.
+//
+// This file is open source and available under the MIT license. See the LICENSE file for more info.
+//
+
+import Foundation
+
+@_spi(AdyenInternal)
+public protocol PresentableInitiableComponent: PresentableComponent, PaymentInitiable {
```

Comment:
Let's find a more fitting name before merging even though it's spi :)
`ConditionallyPresentableInstantComponent` maybe?

Reply:
@goergisn thoughts on this one?

Reply:
My first solution from the top of my head to this naming issue would be to take relief of decision-making out of Swift type system completely.
I'd experiment with one and only one `ComponentPresenting` protocol having single `presentationStyle` property that defines it's behavior. Then during runtime `TwintComponent` or it's controller could easily set it to `requiresPresentation` or `silent`. 

Type-checking any given type for conformance to make business logic decision (or presentation  in this case) is a bit of code smell. Specifically for such a strongly typed language as Swift!